kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/core/v1/PodOperationsImpl.java

```diff
@@ -202,35 +200,28 @@ public Loggable withLogWaitTimeout(Integer logWaitTimeout) {
   @Override
   public PortForward portForward(int port, ReadableByteChannel in, WritableByteChannel out) {
     try {
-      return new PortForwarderWebsocket(httpClient, this.context.getExecutor()).forward(getResourceUrl(), port, in, out);
+      return new PortForwarderWebsocket(httpClient, this.context.getExecutor(), getRequestConfig().getRequestTimeout())
```

Comment:
Same as others, shouldn't we enforce a `0` timeout value?

Reply:
No, the application of the timeout is on a per request basis.  The "default" is 0.

Reply:
Let's take a step back.  The previous enforcement of the request timeout was something that was applied to every request - however it wasn't really applicable to every request, and thus in places we overrode to 0.

What I'm proposing here is that the "default" request timeout from an api perspective is 0 - that's how jetty, jdk, and vertx already operate.  Then only for the operations where the config / context request timeout is applicable, we'll apply it or the related websocket connect timeout.  The reqeustConfig does not need to be further consulted - it was already used to set appropriate timeout.

Reply:
OK, yes, this aligns with what I inferred from the changes. However, aren't the users still capable of overriding the request config for a specific context?

```
client.adapt(NamespacedKubernetesClient.class).withRequestConfig(new RequestConfigBuilder//...
new KubernetesClientBuilder().withConfig(new ConfigBuilder().withRequestTimeout(//...
```

Considering this (maybe I'm still wrong), a user might want to configure some generic timeouts for their HTTP requests, without knowing that they might be actually breaking something else.

Reply:
> However, aren't the users still capable of overriding the request config for a specific context?

Yes, and that requestTimeout would then be applied to applicable requests.

> Considering this (maybe I'm still wrong), a user might want to configure some generic timeouts for their HTTP requests, without knowing that they might be actually breaking something else.

It might be good to clarify that part of the issue is that we have a semantic problem.  The user property is called requestTimeout, while the api level one is a "readTimeout" - for okhttp this means if you don't get anything back from a response stream in that time, then you timeout.  For the other implementations it should be applied for the entire request / response cycle.  

However neither of these is greatly applicable to things like log streaming, which we were overriding at the api level to 0.  That api level setting to 0 would trump anything that user did via the RequestConfig - how this always worked before was that the overriden RequestConfig is logically applied first to create the derived client, then we manipulate that new client instance.  

Am I understanding you correctly that you want new timeout functionality where by a user could use the RequestConfig to override the timeout behavior of following logs (or other operations where we had been explicitly setting it to 0)?

Reply:
> Am I understanding you correctly that you want new timeout functionality where by a user could use the RequestConfig to override the timeout behavior of following logs (or other operations where we had been explicitly setting it to 0)?

Nope, just the opposite, I want to make sure that the user can't override _any_ timeout config in any way that would allow to trip the OpenShift Builds, log requests, watchers, and so on.

I'll probably add some behavioral tests tomorrow on top of this PR. This way I'll probably get a better overview too. There are still things that I can't figure out until I try them out properly. And there also seem to be some remains -or not- (for example the BuildConfigOperationsImpl constructor still has the workaround I added to overcome one of the initial timeout bugs)

Reply:
> for example the BuildConfigOperationsImpl constructor still has the workaround I added to overcome one of the initial timeout bugs

Yes, that should be removed.

Reply:
> Nope, just the opposite, I want to make sure that the user can't override any timeout config in any way that would allow to trip the OpenShift Builds, log requests, watchers, and so on.

Added some changes that clarify the nature of readTimeout as the full request timeout.

The expected behavior:
* the requestTimeout is applied to the HttpRequest via our calling logic, and is not considered separately by the StandardHttpClient.  Whatever is on the HttpRequest is what will be used.
* The Config/RequestConfig.requestTimeout is used for most simple requests via OperationSupport.withRequestTimeout.  That can be overriden pragmatically by NamespacedKubernetesClient.withRequestConfig - which is the only place that OperationContext.withRequestConfig is called from. It is also the only path that calls BaseClient.newClient that uses tag to associate a new RequestConfig with the derived HttpClient.  
* BuildConfigOperationsImpl will use the timeout associated with the context (defaults to 0), the requestConfig does not come into play.
* All of the places that were explicitly using a 0 readTimeout no longer need to do that - that is now the default.